test/dynamo/test_ctx_manager.py

```diff
@@ -22,12 +49,31 @@ def __exit__(self, exc_type, exc_value, traceback):
         torch._C._set_grad_enabled(self.prev)
 
 
+@contextlib.contextmanager
+def customized_ctx_manager(mode):
+    prev = torch.is_grad_enabled()
+    try:
+        yield torch._C._set_grad_enabled(mode)
+    finally:
+        torch._C._set_grad_enabled(prev)
+
+
 class CustomizedCtxManagerWithGraphBreak(CustomizedCtxManager):
     def __enter__(self):
         torch._dynamo.graph_break()
         super().__enter__()
 
 
+@contextlib.contextmanager
+def customized_ctx_manager_with_graph_break(mode):
+    prev = torch.is_grad_enabled()
+    try:
+        torch._dynamo.graph_break()
+        yield torch._C._set_grad_enabled(mode)
+    finally:
+        torch._C._set_grad_enabled(prev)
```

Comment:
Do we have a test for a context manager that does a graph break in the finally: section?

Reply:
Yes! There's one test called `test_graph_break_in_finally` which tests this.